django/utils/autoreload.py

```diff
@@ -222,9 +222,15 @@ def get_child_arguments():
     args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
     # __spec__ is set when the server was started with the `-m` option,
     # see https://docs.python.org/3/reference/import.html#main-spec
-    if __main__.__spec__ is not None and __main__.__spec__.parent:
-        args += ['-m', __main__.__spec__.parent]
-        args += sys.argv[1:]
+    if __main__.__spec__ is not None:
+        modspec = __main__.__spec__
+        if (modspec.name == "__main__" or modspec.name.endswith(".__main__")) and modspec.parent:
+            args += ['-m', modspec.parent]
+            args += sys.argv[1:]
+        else:
+            args += ['-m', modspec.name]
+            args += sys.argv[1:]
```

Comment:
I think a little clearer if less repetitive:
```suggestion
        args.append('-m')
        if (modspec.name == "__main__" or modspec.name.endswith(".__main__")) and modspec.parent:
            args.append(modspec.parent)
        else:
            args.append(modspec.name)
        args += sys.argv[1:]
```